### DISCUSSION TITLE
How can I set metadata for not-found page?

### DISCUSSION THREAD

**Discussion by ziplb** [vercel/next.js#49978] [App Router] [answered]
I created a file `not-found.tsx` in the `app` directory to handle requests for non-existent pages. But it turned out that I can't adapt the page properly, because there is no viewport meta tag. For the same reason, I cannot specify a title for the page.

A static config (`export const metadata = {...}`) or a function that generates it (`generateMetadata`) doesn't help. And this is logical, because the documentation itself [states](https://nextjs.org/docs/app/api-reference/functions/generate-metadata) that they can be specified only for `layout.js` and `page.js files`.

I will be very grateful for your help!

**Comment by mike14747** [comment]
I noticed the same thing and started thinking maybe I should make the title "Page Not Found" in layout.tsx, then make sure I include a title in every page file.

**Reply by ziplb** [reply, marked as answer]
The viewport thing [has been fixed](https://github.com/vercel/next.js/pull/50044). The issue with a title for a page [isn't fixed yet](https://github.com/vercel/next.js/issues/45620). Thanks @GitRowin for the tip 🙌🏻
